packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Utilities/FormattableStringHelpers.cs

```diff
@@ -97,5 +98,130 @@ public static string ReplaceLast(this string text, string oldValue, string newVa
+                result.Add(formattableString);
+            }
+            return result;
+        }
+
+        private static void BreakLinesCore(FormattableString input, StringBuilder formatBuilder, List<object?> args, List<FormattableString> result)
+        {
+            Span<Range> splitIndices = stackalloc Range[input.Format.Length];
+            foreach ((ReadOnlySpan<char> span, bool isLiteral, int index) in StringExtensions.GetFormattableStringFormatParts(input.Format))
+            {
+                // if isLiteral - put in formatBuilder
+                if (isLiteral)
+                {
+                    var numSplits = span.Split(splitIndices, "\n");
+                    for (int i = 0; i < numSplits; i++)
+                    {
+                        var part = span[splitIndices[i]];
+                        // the literals could contain { and }, but they are unescaped. Since we are putting them back into the format, we need to escape them again.
+                        var startsWithCurlyBrace = part.Length > 0 && (part[0] == '{' || part[0] == '}');
+                        var start = startsWithCurlyBrace ? 1 : 0;
+                        var endsWithCurlyBrace = part.Length > 0 && (part[^1] == '{' || part[^1] == '}');
+                        var end = endsWithCurlyBrace ? part.Length - 1 : part.Length;
+                        if (startsWithCurlyBrace)
+                        {
+                            formatBuilder.Append(part[0]).Append(part[0]);
+                        }
+                        if (start <= end) // ensure that we have follow up characters before we move on
+                        {
+                            formatBuilder.Append(part[start..end]);
+                            if (endsWithCurlyBrace)
+                            {
+                                formatBuilder.Append(part[^1]).Append(part[^1]);
+                            }
+                        }
+                        if (i < numSplits - 1)
+                        {
+                            FormattableString formattableString = FormattableStringFactory.Create(formatBuilder.ToString(), args.ToArray());
+                            result.Add(formattableString);
+                            formatBuilder.Clear();
+                            args.Clear();
+                        }
+                    }
+                }
+                // if not Literal, is Args - recurse through Args and check if args has breaklines
+                else
+                {
+                    var arg = input.GetArgument(index);
+                    // when span has a format specifier, go into else case
+                    // if span contains ':' we don't need to split arg and add it directly to FormatBuilder
+                    // TODO: The following logic of the FormatSpecifier handling is temporary until we have a case where FormatSpecifier and \n both exist in an argument.
```

Comment:
How can the format specifier have a line break? That doesn't seem like something we would ever support.

Reply:
it is not the format specifier to have line break. It is about the argument with a line break, and the argument has a format specifier, for example, a case like this:
```
string x = "a\nb";
FormattableString fs = $"content: {x:L}";
```
In our code writer, `:L` means we should generate this in this way:
```
/// content: "a\nb"
```
(maybe we need to escape the `\n` here)
and we have other types of format specifier - this might be a duplicate from the things we have done in CodeWriter, those format specifiers are only defined there.

Reply:
IMO this scenario is unlikely to happen so there is a good chance we will never end up fixing this. I'm fine with having the issue filed, but I don't think we should have a TODO in the code or say that the logic is temporary. We can instead link to the code that would need to be updated in the issue.

Reply:
sure. Then let me remove the TODO here (and maybe organize this better to show the logic here in a better way, now it is a bit messy), and then update the issue content to mark those cases there in case we might forget.